### ISSUE TITLE
Creating a structure named with an existing structure name overwrites the old one

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#3580]
- Binary Ninja Version: 3.2.3782-dev
- OS: Windows 10 x64

**Bug Description:**
Let's say I've this structure:

```c
struct my_struct1
{
    int32_t e1;
    int32_t e2;
    int64_t e3;
};
```

If I create a structure named `my_struct1` by pressing `S`, it warns me about there is already a structure defined with this name, and ask me if I want to overwrite it.

But, If I press `I` and type a structure like this:

```c
struct my_struct1
{
    int32_t bug_test;
};
```

It overwrites currently defined one without asking me.

**Expected Behavior:**
I expect it to ask me just like in other method.

**Comment by CouleeApps** [comment]
This happens because the same code path is used for rename and create types. Proposed solution: remember the operation and warn if you're creating a new type or if you change the name on a rename to a conflict.

**Comment by CouleeApps** [comment]
Looks like the code path for this already existed, it just wasn't checked as of the new parser behavior. This will be fixed in 3.2.3790
